roles/sap_hana_rhv_guest/tasks/set-tuned-profile.yml

```
@@ -0,0 +1,86 @@
+---
+- name: Create tuned profile directory /usr/lib/tuned/sap-hana
+  file:
+    path: /usr/lib/tuned/sap-hana
+    state: directory
+
+- name: Create sap-hana tuned profile
+  copy:
+    dest: "/usr/lib/tuned/sap-hana/tuned.conf"
+    content: |
+        #
+        # tuned configuration
+        #
+        [main]
+        summary=Optimize for SAP HANA
+
+        [cpu]
+        force_latency=cstate.id:3|70
+        governor=performance
+        energy_perf_bias=performance
+        min_perf_pct=100
+
+        [vm]
+        transparent_hugepages=never
+
+        [sysctl]
+        kernel.sem = 32000 1024000000 500 32000
+        kernel.numa_balancing = 0
+        kernel.sched_min_granularity_ns = 3000000
+        kernel.sched_wakeup_granularity_ns = 4000000
+        vm.dirty_ratio = 40
+        vm.dirty_background_ratio = 10
+        vm.swappiness = 10
+
+
+- name: Create tuned profile directory /usr/lib/tuned/sap-hana-kvm-guest
+  file:
+    path: /usr/lib/tuned/sap-hana-kvm-guest
+    state: directory
+
+- name: Add haltpoll.sh for tuned sap-hana-kvm-guest
+  copy:
+    dest: "/usr/lib/tuned/sap-hana-kvm-guest/haltpoll.sh"
+    mode: 0744
+    content: |
+        #!/bin/bash
+
+        if [ "$1" == "start" ]; then
+            modprobe cpuidle-haltpoll force
+        fi
+
+        ## Question:  Does this also need another "if" checking to see if $1 is "stop" to unload the module?
+
+
+- name: Create sap-hana-kvm-guest tuned profile
+  copy:
+    dest: "/usr/lib/tuned/sap-hana-kvm-guest/tuned.conf"
+    content: |
+        #
+        # tuned configuration
+        #
+        [main]
+        summary=Optimize for running SAP HANA on KVM inside a virtual guest
+        include=sap-hana
+
+        [haltpoll]
+        type=script
+        script=${i:PROFILE_DIR}/haltpoll.sh
+
+        [sysfs]
+        /sys/devices/system/clocksource/clocksource0/current_clocksource=tsc
+        /sys/module/haltpoll/parameters/guest_halt_poll_ns=2400000
+        /sys/module/haltpoll/parameters/guest_halt_poll_grow_start=2400000
+
+        [sysctl]
+        kernel.sched_latency_ns=12000000
+        kernel.sched_migration_cost_ns=500000
+        kernel.sched_min_granularity_ns=12000000
+        kernel.sched_wakeup_granularity_ns=15000000
+
+        [bootloader]
+        cmdline_saphana=skew_tick=1
+
+- name: Activate tuned profile
+  command: tuned-adm profile sap-hana-kvm-guest
+...
```

```suggestion
become: true
become_user: root
```
I am not sure but root required here?